### ISSUE TITLE
Locking Connectors

### ISSUE THREAD

**Issue by strikecity** [strikecity/RTSpec#6]
Andrew,

I suggest that we move to the Molex SL 70634 series interconnect family.

For the photodiode board, this would imply the following components:

Surface-Mount Right-Angle 4-Position Locking Connector:
Molex 0015912045
Digikey WM1342-ND

Mating Connector (cable end):
Molex 0050579404
Digikey WM2902-ND

Crimp Pins:
Molex 0016020103
Digikey WM2512-ND

Crimp Tool:
Molex 0638118700
Digikey WM9017-ND

Insertion Tool:
Molex 0011020022
Digikey WM9911-ND

Extraction Tool:
Molex 0011030036
Digikey WM1032-ND

The tools and pins should be the same for all other cables. What we need to determine is:
1) does the surface-mount right-angle version work for the RasPi daughtercard? Or do we need vertical for that? 

2) What are the other pin count(s) that we need? Isn't the LED board just 2 or 3 wires?

**Comment by strikecity** [comment]
Reply from Andrew:

Ryan, to answer your question the LED uses 2 pins. The problem we might encounter is that the LED and photodiode plug into the PCB on the raspberry pi right next to each other. The proximity  between the connectors could cause a problem in fitting it on the board.

**Comment by strikecity** [comment]
Andrew,

For the RasPi I/O board, how about a single shrouded/locking 2 row connector? Then, the cable assembly would "branch" into 2 conductors for the LED and 4 conductors for the photodiode.

Please check whether we have enough clearance for this part.
http://www.digikey.com/product-detail/en/molex-llc/0901301208/WM8248-ND/760960

**Comment by strikecity** [comment]
Here's the 2 position, SMT, R/A connector for the LED board:
Molex: 0015912025
Digikey: WM1340-ND

Mate for the cable assembly:
Molex: 0050579402
Digikey: WM2900-ND

Same crimp pins as above:
Crimp Pins:
Molex 0016020103
Digikey WM2512-ND

**Comment by TheVetter** [comment]
For the header on the RasPi I/O board we will run into the problem of the header being slightly too large for the space. 
For the vertical connector we will won't be able to use the it because it will run into the barrel jack and a few other parts. For the right angle connector we might be able to make it work except it will cover C5 and C2 and we will have to remove R8 and R7 but this isn't an issue because we already remove those two resistors.  

![connector right angle](https://cloud.githubusercontent.com/assets/19328786/20248219/a443fd48-a9a4-11e6-9145-ac45b4dd44c4.PNG)
![new connector vertical](https://cloud.githubusercontent.com/assets/19328786/20248220/a4442a48-a9a4-11e6-9545-96a932b4b81f.PNG)


Right angle header: WM8276-ND
http://www.digikey.com/product-search/en?mpart=0901303208&vendor=900

**Comment by strikecity** [comment]
Good stuff Andrew. I think we should order 2-3 of each type. It's possible that the vertical will just barely squeeze into the space and/or we could melt/cut an offending corner off.

**Comment by TheVetter** [comment]
Ryan,  the cost for the Crimp tool is $345.  I know we don't have much funding for this project and I think the cost will be too high for Dr. Brooks to actually order this tool.

**Comment by TheVetter** [comment]
Ryan, do think that tool specifically is needed or can we get by with one of these tools as a lower cost options.
cheap tool: http://www.digikey.com/product-detail/en/molex-llc/0638111000/WM9999-ND/243789

Medium cost tool: http://www.digikey.com/product-detail/en/molex-llc/0640160201/WM17552-ND/2074098

**Comment by strikecity** [comment]
It's worth a try. Let's try the cheap crimp tool.

I also noticed that, on the RasPi I/O board, we only looked at the board mount connector, not the cable end. And, once I started looking at the cable end + crimp pins, I thought, why not switch EVERYTHING to the Molex C-Grid III family. That way, we can use the same crimp pins for the 2, 4, and 8 position cable terminations. The C-Grid III family supports both single row and double-row.

# Cable terminations:
## 2 position
Molex: 0901560142
Digikey: WM8017-ND

## 4 position
Molex: 0901560144
Digikey: WM8019-ND

## 8 position
Molex: 0901420008
Digikey: WM8036-ND

# PCB Headers:
## 2 position
Molex: 0901362202
Digikey: WM8203-ND

## 4 position
Molex: 0901362204
Digikey: WM8205-ND

## 8 position
Molex: 0901303208
Digikey: WM8276-ND


# Pins and Crimper
## Pins
Molex: 90119-2111
Digikey: WM12409-ND

## Cheap Crimper
Molex: 0638111000
Digikey: WM9999-ND

**Comment by TheVetter** [comment]
Locking connectors arrived today. Hooray! My plan is to install them over the weekend.

**Comment by TheVetter** [comment]
Ryan, Roughly how long should i make the wires between the block and the raspi? Are longer wires going to have more of an antenna effect?

**Comment by strikecity** [comment]
Yes, you are correct that too long could be a problem. But, I think a bit longer than the present system would make it more useable. Maybe 25-50% longer?

**Comment by TheVetter** [comment]
I did not think about this when we ordered the PCB headers but the photodiode and LED PCBs have surface mount connections and the new headers we got are for through hole connections. I was able to make them work but it's not ideal. 
![pd new connector](https://cloud.githubusercontent.com/assets/19328786/20912704/2e12f8b4-bb37-11e6-949c-c1be7c6893f6.jpg)

**Comment by strikecity** [comment]
Andrew,

Good job making these work. The  C-Grid III family does not include SMT header options. We'll have to consider the value of a common connector family vs. the value of SMT headers for LED and photodiode.
